src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -10,72 +9,96 @@ import {
+  // This is a fallback to still find the chapter name if you open the page directly
+  // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
+  if (!page) {
+    page = formPages.find(
+      p => `${formConfig.urlPrefix}${p.path}` === currentPath,
     );
+  }
 
-    let page = eligiblePageList.filter(p => p.path === currentPath)[0];
-    // If the page isn’t active, it won’t be in the eligiblePageList
-    // This is a fallback to still find the chapter name if you open the page directly
-    // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
-    if (!page) {
-      page = formPages.find(
-        p => `${formConfig.urlPrefix}${p.path}` === currentPath,
-      );
+  let current;
+  let chapterName;
+  if (page) {
+    current = chapters.indexOf(page.chapterKey) + 1;
+    // The review page is always part of our forms, but isn’t listed in chapter list
+    chapterName =
+      page.chapterKey === 'review'
+        ? formConfig?.customText?.reviewPageTitle || REVIEW_APP_DEFAULT_MESSAGE
+        : formConfig.chapters[page.chapterKey].title;
+    if (typeof chapterName === 'function') {
+      chapterName = chapterName();
     }
+  }
+
+  const stepText = `Step ${current} of ${chapters.length}: ${chapterName}`;
+  const showHeader = Math.abs(current - index) === 1;
 
-    let current;
-    let chapterName;
-    if (page) {
-      current = chapters.indexOf(page.chapterKey) + 1;
-      // The review page is always part of our forms, but isn’t listed in chapter list
-      chapterName =
-        page.chapterKey === 'review'
-          ? formConfig?.customText?.reviewPageTitle ||
-            REVIEW_APP_DEFAULT_MESSAGE
-          : formConfig.chapters[page.chapterKey].title;
-      if (typeof chapterName === 'function') {
-        chapterName = chapterName();
+  // The goal with this is to quickly "remove" the header from the DOM, and
+  // immediately re-render the component with the header included. This should
+  // ensure that VoiceOver on iOS will pick up on the new <h2>
+  useEffect(
+    () => {
+      if (current > index + 1) {
+        setIndex(index + 1);
+      } else if (current === index) {
+        setIndex(index - 1);
       }
-    }
 
-    const stepText = `Step ${current} of ${chapters.length}: ${chapterName}`;
-
-    return (
-      <div>
-        <SegmentedProgressBar total={chapters.length} current={current} />
-        <div className="schemaform-chapter-progress">
-          <div
-            aria-valuenow={current}
-            aria-valuemin="1"
-            aria-valuetext={stepText}
-            aria-valuemax={chapters.length}
-            className="nav-header nav-header-schemaform"
-          >
-            <h2 className="vads-u-font-size--h4">{stepText}</h2>
-          </div>
+      return () => {
+        focusElement('.nav-header > h2');
+      };
+    },
+    [current, index],
+  );
```

Comment:
Anytime `current` changes, we want to update the `index` state variable appropriately, depending on if we are going forward to a new chapter or backward to a previous one. Updating the `index` will also trigger a re-render, and it is this follow-up re-render that causes the `<h2>` to be rendered instead of the placeholder `<div>`.

We return the `focusElement` call in the cleanup function to ensure that focus remains where we want it after we're done doing this special re-rendering of things so that VoiceOver will notice the change.